compiler/test/fail_compilation/invalid_lib.d

```diff
@@ -9,7 +10,7 @@ Use a regex because the path is really strange on Azure (OMF_32, 64):
 
 TEST_OUTPUT:
 ----
-$r:.*$: Error: corrupt $?:windows=MS Coff|osx=Mach|ELF$ object module $?:windows=fail_compilation\extra-files\fake.lib|fail_compilation/extra-files/fake.a$ $n$
+Error: corrupt ELF object module fail_compilation/extra-files/fake.a 112
```

Comment:
If this PR is a refactor, why is the test changed? Did the behavior change?

Reply:
I did not understand the test case. There are no comments in it.

Reply:
It's testing that linking with a corrupt library generates the correct message, which depends on the platform. Now there is no test for Mac or Windows anymore. Can't you leave it as it is?

Reply:
It failed as is, saying it got the green message instead of the red one.

Reply:
I see, it's because you now pass `Loc.initial` to the error functions instead of `this.loc`. Please add the correct location back, that's useful information.